### PR TITLE
Compose tls_secret_name according to kube-lego availability

### PR DESCRIPTION
**Pull request by romanovskyj**

### CODE REVIEW

File: apsconnectcli/apsconnect.py [hunk 1]
```diff
@@ -844,7 +844,17 @@ def _create_ingress(hostname, name, api, ext_api, namespace='default',
 
     if not tls_secret_name:
         generated_tls = True
-        tls_secret_name = '{}-tls'.format(hostname)
+
+        try:
+            api.read_namespace('kube-leg')
```
Comment (ToxicWar) [inline apsconnectcli/apsconnect.py]: fix typo `kube-lego`

File: apsconnectcli/apsconnect.py [hunk 2]
```diff
@@ -844,7 +844,17 @@ def _create_ingress(hostname, name, api, ext_api, namespace='default',
...
+            tls_secret_name = '{}-tls'.format(hostname)
+        except ApiException as e:
+            if e.status == 404:
+                # Hostname without a subdomain part
+                hostname_domain = ('.').join(hostname.split('.')[1:])
```
Comment (ToxicWar) [inline apsconnectcli/apsconnect.py]: > 'test.my.domain.example.com'.split('.', 1)
['test', 'my.domain.example.com']

File: apsconnectcli/apsconnect.py [hunk 3]
```diff
@@ -844,7 +844,17 @@ def _create_ingress(hostname, name, api, ext_api, namespace='default',
...
-        tls_secret_name = '{}-tls'.format(hostname)
+
+        try:
+            api.read_namespace('kube-lego')
+            tls_secret_name = '{}-tls'.format(hostname)
+        except ApiException as e:
+            if e.status == 404:
+                # Hostname without a subdomain part
+                _, hostname_domain = hostname.split('.', 1)
+                tls_secret_name = 'star.{}'.format(hostname_domain)
```
Comment (ToxicWar) [inline apsconnectcli/apsconnect.py]: Don't drop this secret from k8s cluster or remove logic when we drop `tls` secret at all

Comment (romanovskyj) [reply apsconnectcli/apsconnect.py]: Done

**Pull request merged**